### PR TITLE
Release v5.9.1

### PR DESCRIPTION
**Pull request by tis24dev**
Automated release PR for `v5.9.1`.

## Summary by Sourcery

Align range parameter syncing with device shadows by snapping off-grid shadow values onto the range grid so commands reflect the actual setpoints, and bump the integration version to v5.9.1.

New Features:
- Add a snap_to_grid helper on range parameters to derive the nearest in-range grid value from an off-grid input.

Bug Fixes:
- Ensure sync_params_to_command snaps off-grid shadow values for range parameters instead of falling back to default minimums, preventing unintended setpoint resets.

Tests:
- Add unit tests for snap_to_grid behavior on range parameters, including bounds handling and setter acceptance.
- Add a regression test for syncing off-grid shadow range values into commands to verify correct snapping and prevent default fallback.

Chores:
- Update the addhon integration manifest version from 5.9.0 to 5.9.1.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved synchronization of appliance settings when values are missing, unknown, or not aligned with permitted increments.
  * Off-grid numeric values are now adjusted to the nearest valid setting, while values outside allowed limits are clamped safely.
  * Added clearer handling when a setting value cannot be applied.

* **Tests**
  * Added coverage for grid snapping, boundary values, and shadow-setting synchronization.

* **Chores**
  * Updated the integration version to 5.9.1.

<h3>Greptile Summary</h3>

This PR aligns range parameter syncing with device shadows and releases version 5.9.1. The main changes are:

- Snap in-range, off-grid shadow values to valid range steps.
- Reject out-of-range shadow values instead of clamping them.
- Add tests for snapping, bounds, parsing, and command syncing.
- Bump the integration version to 5.9.1.

<h3>Confidence Score: 5/5</h3>

No additional blocking issue qualifies for this follow-up review.

- The updated range setter path preserves its stored value when validation fails.
- The snapping helper returns values accepted by the existing range validation.
- No separate production failure remained after applying the follow-up selection scope.

<sub>Reviews (2): Last reviewed commit: ["fix(engine): don&#39;t clamp out-of-range sh..."](https://github.com/tis24dev/addhon/commit/5b74e14b73caf516feb82e4e60093bdee0c309c6) | [Re-trigger Greptile](https://app.greptile.com/api/retrigger?id=46109577)</sub>

### CODE REVIEW

File: custom_components/addhon/client/engine/appliance.py [hunk 1]
```diff
@@ -374,7 +378,20 @@ def sync_params_to_command(self, command_name: str) -> None:
...
+                # The device shadow can report an OFF-GRID value for a range setpoint
+                # (a measured 17.2 for a step-1 target). Skipping would leave the command
+                # at its load-time default (min), which a later FULL-command send writes
+                # back to the device -- clobbering the real setpoint (both wine-cooler
+                # zones snapped to 5C when the light switch fired, discussion #62). Snap
+                # the shadow value onto the grid so the command carries the true setpoint.
+                snap = getattr(setting, "snap_to_grid", None)
+                if callable(snap):
+                    try:
+                        setting.value = snap(raw)
```
[the remark this answers is not in the document]
Comment (tis24dev) [reply custom_components/addhon/client/engine/appliance.py:392]: Good catch — fixed in 5b74e14. snap_to_grid now raises ValueError when the value is outside [min, max] instead of clamping it, so sync_params_to_command skips it and keeps the command's current (possibly user-set) value rather than overwriting it with a boundary guess from a stale/out-of-range reading. Only in-range off-grid values (the #62 case, e.g. 17.2 in 5..20) are snapped. Added a regression: an out-of-range shadow (25) leaves a user-set command value (18) untouched.

Note on the trade-off: at cold start (command still at its load-time default) an out-of-range shadow is now skipped and the command keeps the default, matching the pre-fix behavior. That path is unreachable on a well-formed device (the command range is the device's real capability, so a shadow setpoint is always in range), and preserving an already-good command value is the correct general rule.

File: tests/test_engine_parameters.py [hunk 2]
```diff
@@ -286,6 +286,54 @@ def test_values_bounded_on_malformed_range(self) -> None:
...
+    def _range(self, lo, hi, step):
+        return NaRange("temp", {"category": "command", "typology": "range",
+                                "mandatory": 0, "minimumValue": lo, "maximumValue": hi,
+                                "incrementValue": step, "defaultValue": lo}, "grp")
+
+    def test_snaps_off_grid_to_nearest(self) -> None:
+        p = self._range("5", "20", "1")
+        self.assertEqual(p.snap_to_grid("17.2"), 17)   # the #62 case
+        self.assertEqual(p.snap_to_grid("12.2"), 12)
+        self.assertEqual(p.snap_to_grid(17.6), 18)     # rounds to nearest
```
[the remark this answers is not in the document]
Comment (tis24dev) [reply tests/test_engine_parameters.py:304]: Added in 5b74e14: test_non_numeric_raises asserts snap_to_grid("bad") raises ValueError. The out-of-range tests were also switched to assertRaises (snap now raises rather than clamps), so the ValueError behavior on invalid input is pinned.
